### PR TITLE
List in NodesInterface takes label selector

### PR DESCRIPTION
**Pull request by kazegusuri**

### CODE REVIEW

File: pkg/client/nodes_test.go [hunk 1]
```diff
@@ -38,10 +41,41 @@ func TestListMinions(t *testing.T) {
...
+						},
+					},
+				},
+			},
+		},
+	}
+	c.Setup()
+	c.QueryValidator[labelSelectorQueryParamName] = validateLabels
+	selector := labels.Set{"foo": "bar", "name": "baz"}.AsSelector()
+	receivedPodList, err := c.Nodes().List(selector)
```
Comment (davidopp) [inline pkg/client/nodes_test.go]: I know this is a dumb question, but where does the list of fake nodes get populated? IIUC the List() being called here is the one in fake_minions.go, which returns &c.Fake.MinionsList, but I couldn't find where that MinionsList actually gets populated.

Comment (kazegusuri) [inline pkg/client/nodes_test.go]: IIUC this test uses mock http server, not fake clients. So this checks labels are passed via request parameters.

Comment (davidopp) [inline pkg/client/nodes_test.go]: Sorry I'm being dense, but I still don't understand. Is the List() that's invoked here the one in fake_minions.go? If so, where does the list of nodes that gets returned get populated?

**Comment by davidopp** [comment]
LGTM

I don't want to hold up merging this, but it would be great if you could answer my question from the last review. I'll merge it now.

**Comment by davidopp** [comment]
Ah, sorry, looks like it needs a rebase. Please ping when it is rebased.

File: pkg/client/nodes_test.go (hunk 1 above)
Comment (kazegusuri) [inline pkg/client/nodes_test.go]: Sorry for delay. This doesn't use fake_minions.go at all. This calls List() in nodes.go, so it sends a request to mocked apiserver.

**Comment by kazegusuri** [comment]
@davidopp Sorry for my delayed response. PTAL.

**Comment by kazegusuri** [comment]
Rebased. ping @davidopp

**Pull request merged**